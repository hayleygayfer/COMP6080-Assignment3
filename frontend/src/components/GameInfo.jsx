import React from 'react';
import PropTypes from 'prop-types';
function GameInfo (props) {
  return <>
  Title: {props.title}<br/>
  Thumbnail: {props.thumbnail ? <img src={props.thumbnail}> </img> : 'none provided' } <br/>
  </>
}
export default GameInfo;
GameInfo.propTypes = {
  title: PropTypes.string, // gameData[i].name
  thumbnail: PropTypes.string // gameData[i].thumbnail
};
