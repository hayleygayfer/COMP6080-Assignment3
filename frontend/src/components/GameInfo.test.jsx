import { shallow } from 'enzyme';
import * as React from 'react';
import GameInfo from './GameInfo'
import renderer from 'react-test-renderer';

describe('GameInfo', () => {  
  // const mockCallBack = jest.fn();

  it('uses custom title', () => {
    const customTitle = 'RATTLEBONE\'S GAME';
    const gameInfo = shallow(<GameInfo title={customTitle} thumbnail={null}/>)
    expect(gameInfo.text()).toBe('Title: RATTLEBONE\'S GAMEThumbnail: none provided ') // check if text matches
  })

  it('uses custom image src', () => {
    const customImage = 'https://i.pinimg.com/236x/5c/d2/c8/5cd2c8ccbce5f05e3a1023d94b7ee732.jpg';
    const gameInfo = shallow(<GameInfo thumbnail={customImage}/>)
    expect(gameInfo.find('img').prop('src')).toEqual(customImage);
  })

  it('handles no thumbnail', () => {
    const gameInfo = shallow(<GameInfo thumbnail={null}/>)
    expect(gameInfo.text()).toBe('Title: Thumbnail: none provided ') // check if text matches
    // expect(button.text()).toBe(customText) // check if text matches
  })

  // SNAPSHOT TESTING....
  it('uses custom title (snapshot)', () => {
    const customTitle = 'RATTLEBONE\'S GAME';
    const gameInfo = renderer.create(<GameInfo title={customTitle} thumbnail={null}/>);
    expect(gameInfo).toMatchSnapshot();
  })

  it('uses custom image src (snapshot)', () => {
    const customImage = 'https://i.pinimg.com/236x/5c/d2/c8/5cd2c8ccbce5f05e3a1023d94b7ee732.jpg';
    const gameInfo = renderer.create(<GameInfo thumbnail={customImage}/>)
    expect(gameInfo).toMatchSnapshot();
  })

  it('handles no thumbnail (snapshot)', () => {
    const gameInfo = renderer.create(<GameInfo thumbnail={null}/>)
    expect(gameInfo).toMatchSnapshot();
  })
});
