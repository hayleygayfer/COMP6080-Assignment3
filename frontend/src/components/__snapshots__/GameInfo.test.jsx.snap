// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo handles no thumbnail (snapshot) 1`] = `
Array [
  "Title: ",
  <br />,
  "Thumbnail: ",
  "none provided",
  " ",
  <br />,
]
`;

exports[`GameInfo uses custom image src (snapshot) 1`] = `
Array [
  "Title: ",
  <br />,
  "Thumbnail: ",
  <img
    src="https://i.pinimg.com/236x/5c/d2/c8/5cd2c8ccbce5f05e3a1023d94b7ee732.jpg"
  >
     
  </img>,
  " ",
  <br />,
]
`;

exports[`GameInfo uses custom title (snapshot) 1`] = `
Array [
  "Title: ",
  "RATTLEBONE'S GAME",
  <br />,
  "Thumbnail: ",
  "none provided",
  " ",
  <br />,
]
`;
